fb2k/foo_streamfile.cpp

```diff
@@ -193,16 +199,23 @@ static STREAMFILE * open_foo_streamfile_buffer_by_file(service_ptr_t<file> m_fil
 static STREAMFILE * open_foo_streamfile_buffer(const char * const filename, size_t buffersize, abort_callback * p_abort, t_filestats * stats) {
     STREAMFILE *streamFile;
     service_ptr_t<file> infile;
+    bool infile_exists;
 
-    if(!(filesystem::g_exists(filename, *p_abort)))
-        return NULL;
+    infile_exists = filesystem::g_exists(filename, *p_abort);
+    if(!infile_exists) {
+        /* allow non-existing files in some cases */
+        if (!vgmstream_is_virtual_filename(filename))
+            return NULL;
+    }
 
-    filesystem::g_open_read(infile,filename,*p_abort);
-    if(stats) *stats = infile->get_stats(*p_abort);
+    if (infile_exists) {
+        filesystem::g_open_read(infile,filename,*p_abort);
+        if(stats) *stats = infile->get_stats(*p_abort);
+    }
 
-    streamFile = open_foo_streamfile_buffer_by_file(infile,filename,buffersize,p_abort);
+    streamFile = open_foo_streamfile_buffer_by_file(infile, infile_exists, filename, buffersize, p_abort);
     if (!streamFile) {
-        // fclose(infile);
+        //m_file.release(); //todo not needed after g_open_read?
```

Comment:
service_ptr_t is reference counting and auto releasing when it goes out of scope. Manual releasing is only necessary when it's a member of a structure that wasn't allocated with new. Gotta make sure it's initialized in place in that case.